### PR TITLE
[Merged by Bors] - feat: generalize `Int.cast_nonneg` and related lemmas

### PR DESCRIPTION
**Pull request by j-loreaux**
---

If this benchmarks fine, I think it's a good change. See also [this comment](https://github.com/leanprover-community/mathlib4/pull/13974#discussion_r1680054076) on #13974

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by j-loreaux** [comment]
!bench

File: Mathlib/Algebra/Order/Ring/Cast.lean [hunk 1]
```diff
@@ -25,14 +25,16 @@ variable {R : Type*}
 
 namespace Int
 section OrderedRing
```
Comment (riccardobrasca) [inline Mathlib/Algebra/Order/Ring/Cast.lean:27]: Can you change the name of the section please?

**Review by riccardobrasca — APPROVED** [review]
Does it make sense to move this to another file? Maybe it is not worth the trouble, so don't worry too much about this.

Thanks!

bors d+

**Comment by j-loreaux** [comment]
bors merge

**Comment by j-loreaux** [comment]
> Does it make sense to move this to another file? Maybe it is not worth the trouble, so don't worry too much about this.

I don't think it's worth it.

**Pull request closed without merging**